### DISCUSSION TITLE
Solving Matrix equations

### DISCUSSION THREAD

**Discussion by andreichalapco** [sympy/sympy#24506] [Q&A]
When trying to do this:

```python
from sympy import *
A = MatrixSymbol('A', n, n)
x = MatrixSymbol('x', n, 1)
b = MatrixSymbol('b', n, 1)
solve(A*x - b, x)
```python

```shell
TypeError: First argument of MatrixElement should be a matrix
```

I was expecting something in the lines of x = A**(-1) *b

**Comment by 1e9abhi1e10** [comment]
I think  `A*X ` and  `-b` are not aligned

**Comment by Vandermode** [comment]
I am facing similar issues. It seems the solver for matrix calculus is unmatured. 
For example, consider a simple least square problem in matrix form:
```python
from sympy import *
A = MatrixSymbol('A', n, n)
x = MatrixSymbol('x', n, 1)
b = MatrixSymbol('b', n, 1)
r = A*x - b
lsq_obj = r.T * r
```
we want to minimize the least square objective by setting the first derivative to 0
```python
solve(lsq_obj.diff(x), x)
```
However, the above code doesn't work properly, returning None or errors.
